engine/engine/content/builtins/scripts/debugger.lua

```diff
@@ -63,28 +63,30 @@ local function remove_breakpoint(line, file)
   end
 end
 
+local file_cache = {}
```

Comment:
Usually project has not so many files to process + some default Lua sources like `[C]`. Processing of each string is pretty heavy operation, so it's much faster (x2+) to cache the result